src/library/dlgcoverartfullsize.cpp

```diff
@@ -130,8 +132,9 @@ void DlgCoverArtFullSize::slotCoverFound(const QObject* pRequestor,
         // whitespace appearing on the side when resizing a window whose
         // borders touch the edges of the screen.
         QSize dialogSize = m_pixmap.size();
-        const QSize availableScreenSpace =
-            QApplication::desktop()->availableGeometry().size() * 0.9;
+        const QList<QScreen*> screens = QGuiApplication::screens();
+        DEBUG_ASSERT(!screens.isEmpty());
```

Comment:
This list should never be empty, but we cannot rely on this assumption. The Qt docs do not promise anything. Better use VERIFY_OR_DEBUG_ASSERT with a warning (or critical?) log message and either fall back to some default behavior/settings or abort the operation.

Reply:
Maybe encapsulated in a getPrimaryScreen() utility function that reveals the intention instead of just using a low-level function like QList::first().

Reply:
> Maybe encapsulated in a getPrimaryScreen() utility function that reveals the intention instead of just using a low-level function like QList::first().

I re-checked the Qt docs and found [QGuiApplication's primaryScreen property](https://doc.qt.io/qt-5/qguiapplication.html#primaryScreen-prop). This works for Qt >= 5.6. Is mixxx compatible with older Qt5 versions? Also, this property isn't static, so we'd need to pass around a reference to the QtGuiApplication instance.

Reply:
Currently the minimum supported version is Qt 5.4.

Reply:
Since `QScreen` has no public constructor, I decided against `getPrimaryScreen` and used `VERIFY_OR_DEBUG_ASSERT`. Please have a look at the respective commits and tell if the implementation is okay.